src/gui/folderman.cpp

```diff
@@ -1588,25 +1589,40 @@ Folder *FolderMan::addFolder(AccountStatePtr accountStatePtr, const QString &loc
         folderDefinition.navigationPaneClsid = QUuid::createUuid();
 #endif
 
-    auto f = addFolder(accountStatePtr, folderDefinition);
+    auto finalize = [this, accountStatePtr, syncMode, localFolder](const FolderDefinition &folderDefinition, const QStringList &blacklist = {}) {
+        auto f = addFolder(accountStatePtr, folderDefinition);
 
-    if (f) {
-        if (folderDefinition.virtualFilesMode != Vfs::Off && OwncloudWizard::useVirtualFileSync())
-            f->setRootPinState(PinState::OnlineOnly);
+        if (f) {
+            if (folderDefinition.virtualFilesMode != Vfs::Off && syncMode == Wizard::SyncMode::UseVfs)
+                f->setRootPinState(PinState::OnlineOnly);
+
+            f->journalDb()->setSelectiveSyncList(SyncJournalDb::SelectiveSyncBlackList, blacklist);
 
-        f->journalDb()->setSelectiveSyncList(SyncJournalDb::SelectiveSyncBlackList,
-            OwncloudWizard::selectiveSyncBlacklist());
-        if (!OwncloudWizard::isConfirmBigFolderChecked()) {
-            // The user already accepted the selective sync dialog. everything is in the white list
-            f->journalDb()->setSelectiveSyncList(SyncJournalDb::SelectiveSyncWhiteList,
-                QStringList() << QLatin1String("/"));
+            if (!OwncloudWizard::isConfirmBigFolderChecked()) {
+                // The user already accepted the selective sync dialog. everything is in the white list
+                f->journalDb()->setSelectiveSyncList(SyncJournalDb::SelectiveSyncWhiteList,
+                    QStringList() << QLatin1String("/"));
+            }
+            qCDebug(lcFolderMan) << "Local sync folder" << localFolder << "successfully created!";
+        } else {
+            qCWarning(lcFolderMan) << "Failed to create local sync folder!";
         }
-        qCDebug(lcFolderMan) << "Local sync folder" << localFolder << "successfully created!";
+
+        return f;
+    };
+
+    if (syncMode == Wizard::SyncMode::SelectiveSync) {
+        auto dialog = new SelectiveSyncDialog(accountStatePtr->account(), remotePath, reinterpret_cast<QDialog *>(ocApp()->gui()->settingsDialog()));
```

Comment:
static_cast or qobject_cast.
Why cast at all? To save one include?

Reply:
The compiler was complaining because the constructor is `explicit`.